test/logger/POSTLogger.test.ts

```diff
@@ -0,0 +1,279 @@
+
+describe('POSTLogger', () => {
+  const expect: Chai.ExpectStatic = chai.expect;
+  let domMockBuilder: DOMMockBuilder | null = null;
+  let domMockBehavior: DOMMockBehavior | null = null;
+  const BASE_URL = 'base_url';
+  const intervalMs = 50;
+  const batchSize = 2;
+
+  beforeEach(() => {
+    domMockBehavior = new DOMMockBehavior();
+    domMockBuilder = new DOMMockBuilder(domMockBehavior);
+  });
+
+  after(() => {
+    if (domMockBuilder) {
+      domMockBuilder.cleanup();
+      domMockBuilder = null;
+    }
+  });
+
+  describe('construction', () => {
+    it('can be constructed', async () => {
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.WARN);
+      expect(logger).to.not.equal(null);
+      expect(logger.getLogLevel()).to.equal(LogLevel.WARN);
+      await logger.destroy();
+    });
+
+    it('can be with different level', async () => {
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.DEBUG);
+      expect(logger).to.not.equal(null);
+      expect(logger.getLogLevel()).to.equal(LogLevel.DEBUG);
+      await logger.destroy();
+    });
+
+    it('can be constructed with optional metadata', async () => {
+      const options = {
+        metadata: { MeetingId: '12345' },
+      };
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.INFO, options);
+      expect(logger).to.not.equal(null);
+      expect(logger.getLogLevel()).to.equal(LogLevel.INFO);
+      expect(logger.metadata['MeetingId']).to.exist;
+      expect(logger.metadata['MeetingId']).to.equal('12345');
+      await logger.destroy();
+    });
+  });
+
+  describe('HTTP POST metadata', () => {
+    it('can get and set metadata', async () => {
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.INFO);
+      expect(logger).to.not.equal(null);
+      expect(logger.getLogLevel()).to.equal(LogLevel.INFO);
+      expect(logger.metadata).to.be.undefined;
+      logger.metadata = { MeetingId: '12345' };
+      expect(logger.metadata['MeetingId']).to.equal('12345');
+      logger.metadata = { MeetingId: '45678', AttendeeId: '12345' };
+      expect(logger.metadata['MeetingId']).to.equal('45678');
+      expect(logger.metadata['AttendeeId']).to.equal('12345');
+      await logger.destroy();
+    });
+  });
+
+  describe('disposal', () => {
+    it('can be disposed', async () => {
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.WARN);
+      await logger.destroy();
+      // This is safe to call twice.
+      await logger.destroy();
+    });
+  });
+
+  describe('logging level', () => {
+    it('should log info with LogLevel.INFO and ignore the debug', async () => {
+      const logger = new POSTLogger(batchSize, intervalMs, BASE_URL, LogLevel.INFO);
+      logger.info('info');
+      logger.error('error');
+      logger.debug('error');
+      expect(logger.getLogCaptureSize()).is.equal(2);
+      await wait(1000);
```

Comment:
Any reason it has to be that big? We should try to do min wait time if possible.

Reply:
I have removed the wait from most places where sending is not actually checked, and updated other places based on the logs and time. Thanks and this is good point to change!